src/utils/useVideoTiming.ts

```diff
@@ -0,0 +1,97 @@
+import { useState, useEffect } from "react";
+import { TARGET_TYPE } from "../constants";
+import { getVideoCurrentTimeInMilliseconds } from "./util";
+
+export interface UseVideoTimingProps {
+    targetType: typeof TARGET_TYPE.PAGE | typeof TARGET_TYPE.FRAME;
+    referenceEl?: HTMLElement;
+    activeAnnotationId: string | null;
+    annotations: Array<{ id: string; target: { location: { value: number } } }>;
+}
+
+export interface UseVideoTimingReturnType {
+    isVideoSeeking: boolean;
+    getCurrentVideoLocation: () => number;
+}
+
+const useVideoTiming = ({
+    targetType,
+    referenceEl,
+    activeAnnotationId,
+    annotations,
+}: UseVideoTimingProps): UseVideoTimingReturnType => {
+    const [isVideoSeeking, setIsVideoSeeking] = useState<boolean>(false);
+    const [targetVideoTime, setTargetVideoTime] = useState<number | null>(null);
```

Comment:
Can we get rid of the `null` type here and just use `undefined` when there is no `targetVideoTime`?

Reply:
Null is actually a valid value. In the handleTimeUpdate method we set it to null if it's reached the target time within 100ms. I like the explicit assignment here as null actually means something in this context.